### DISCUSSION TITLE
Allure Cucumbers.js Integration: Error with allure.step or attachment - 'test.info() can only be called while test is running

### DISCUSSION THREAD

**Discussion by sujith0980** [allure-framework/allure2#2824] [Questions & Support] [answered]
I have created a project with CucumberJs and Allure. So importing allure from allure-js-commons and writing steps throwing an error.
**Error: test.info() can only be called while test is running**

import { Given, Then, When, setDefaultTimeout } from "@cucumber/cucumber";
import { loginPage } from "../../Object-Repository/login.mjs";
import { attachment } from "allure-js-commons";

Given("I am on the login page", async function () {
  await attachment(
    "Sample text attachment for step",
    "Goodbye world!",
    "text/plain"
  );
  await await global.page.goto(
    "https://youtube.com"
  );
  await global.page.fill(loginPage.usernameField, "admin");
  await global.page.click(loginPage.submitButton);
  await global.page.fill(loginPage.credentialsField, "Admin");
  await global.page.click(loginPage.loginButton);
  console.log("Logged into Application");
});

Could anyone help me out with this issue?

**Comment by baev** [comment]
Please provide more details on your setup. It looks like you're using CucumberJS runner with Playwright as a library. But according to the error, you're trying to use the `allure-playwright` dependency, which requires a Playwright runner.

**Comment by sujith0980** [comment]
Its my cucumber.json

{
    "default": {
        "paths": [
            "tests/features/*.feature"
            
        ],
        "require": [
            "tests/steps/*.js",
            "tests/hooks/hooks.js"
        ],

        "formatOptions": {
            "snippetInterface": "async-await",
            "resultsDir": "allure-results"
        },
        "format": [
            ["allure-cucumberjs/reporter"]
        ],
}

As you provided in the allure cucumberjs official document.  I have tried allure.step but got the same error. 

its my package.json

"devDependencies": {
        "@cucumber/cucumber": "^10.9.0",
        "@cucumber/messages": "^27.0.2",
        "@cucumber/pretty-formatter": "^1.0.1",
        "@playwright/test": "^1.49.0",
        "allure-commandline": "^2.32.0",
        "allure-cucumberjs": "^3.0.7",
    },

"dependencies": {
        "allure-js-commons": "^3.0.7",
        "allure-playwright": "^3.0.6",
    }
    
    
And nothing else. Just before and after hooks to open and close the browser. And these BeforeAll, Before, After and AfterAll hooks not being considered by allure. In the allure report I can only see the test body. Setup and Teardown sections are empty.

**Reply by baev** [reply]
You need to remove `allure-playwright` dependency. It makes Allure think you're using Playwright Runner

**Comment by sujith0980** [comment]
**I have removed allure-playwright from the package.json**

```
"devDependencies": {
        "@babel/core": "^7.26.0",
        "@babel/preset-env": "^7.26.0",
        "@babel/register": "^7.25.9",
        "@cucumber/cucumber": "^10.9.0",
        "@cucumber/messages": "^27.0.2",
        "@cucumber/pretty-formatter": "^1.0.1",
        "@playwright/test": "^1.49.0",
        "@types/node": "^22.9.1",
        "allure-commandline": "^2.32.0",
        "allure-cucumberjs": "^3.0.7",

    },
    "allure": {
        "outputDir": "./allure-results",
    },
    "dependencies": {
        "allure-js-commons": "^3.0.7"
    }
```
    
**this is sample hooks im using**
    
```
    import * as allure from "allure-js-commons";

//before all feature files
BeforeAll(async function () {
  console.log("Running");
  await allure.description("It has something to do");
  browser = await chromium.launch({
    headless: false,
    slowMo: 20,
    timeout: 80 * 1000,
    args: ["--start-maximized"],
  });
});
```

**This is the error im getting if I just remove allure-playwright from the package.json**

Error: a BeforeAll hook errored, process exiting: tests\hooks\hooks.js:14
    at Runtime.runTestRunHooks (D:\Sample\node_modules\@cucumber\cucumber\src\runtime\run_test_run_hooks.ts:37:19)
[cause]: Error: test.info() can only be called while test is running
at AllurePlaywrightTestRuntime.apply (D:\Sample\node_modules\allure-playwright\src\runtime.ts:2:1)
      at AllurePlaywrightTestRuntime.sendMessage (D:\Sample\node_modules\allure-playwright\src\runtime.ts:29:4)
at AllurePlaywrightTestRuntime.description (D:\Sample\node_modules\allure-js-commons\src\sdk\runtime\MessageTestRuntime.ts:73:8)
      at D:\Sample\node_modules\allure-js-commons\src\facade.ts:26:31
      
**And I tried uninstalling allure-playwright and deleted allure-playwright from  node_modules and this is the error I got**

Error: a BeforeAll hook errored, process exiting: tests\hooks\hooks.js:14
    at Runtime.runTestRunHooks (D:\Sample\node_modules\@cucumber\cucumber\src\runtime\run_test_run_hooks.ts:37:19)     
 [cause]: Error: Cannot find package 'allure-playwright' imported from D:\Sample\node_modules\allure-js-commons\dist\cjs\sdk\runtime\runtime.js
code: 'ERR_MODULE_NOT_FOUND'

**Reply by baev** [reply, marked as answer]
You will have such an effect if you don't import `allure-cucumberjs` in cucumber world; see https://github.com/allure-examples/cucumberjs-npm-esm-ts/blob/main/features/support/world.ts

**Reply by sujith0980** [reply]
Oh great..!! It worked when I placed the world.js in my tests/steps folder.
1.  The thing is I have created hooks BeforeAll, Before, After and AfterAll and I ma expecting that allure takes these as Setup and Teardown but in allure report I see Setup and teardown sections are empty. Any help regarding this?
2. Why cannot I use Formatter as summary or progress-bar and allure-cucumberjs/reporter together?

**Reply by baev** [reply]
> 2\. Why cannot I use Formatter as summary or progress-bar and allure-cucumberjs/reporter together?

see https://github.com/orgs/allure-framework/discussions/2739#discussioncomment-10729259

**Comment by arcaneGamer888** [comment]
I also tried this but it is throwing
 Error: Attempted to access `world` from incorrect scope; only applicable to steps and case level hooks.

the rough structure of my framework is; 
foldername
|
src
|>hooks
--->    hooks.ts
--->    world.ts  (import "allure-cucumberjs")
|>test
----|>features
---------->demo.feature (inside feature folder)
 ----|>steps
---------->demoSteps.ts (inside steps folder)

cucumber.js
package.json


the versions are ;
  "allure-cucumberjs": "^3.0.6",
  "allure-js-commons": "^3.0.6",
"allure-commandline;:"^2.32.0"

**Reply by baev** [reply]
@arcaneGamer888, please create a separate discussion. Make sure to provide additional details on your setup

**Reply by yakomoga** [reply]
I have this same error when  adding attachment to a failed step and I don't use custom world.
The report itself generates correctly but without attachment.

**Reply by baev** [reply]
@yakomoga please create a new discussion and describe your issue

**Reply by yakomoga** [reply]
> @yakomoga please create a new discussion and describe your issue

I created a [new discussion](https://github.com/orgs/allure-framework/discussions/2880) with the link to a minimal repo to reproduce the problem
